### PR TITLE
Add Codecov support in Github Actions.

### PR DESCRIPTION
**Pull request by HyukjinKwon**
Codecov in Github Actions are not correctly working for now:

```
...
==> GitHub Actions detected.
    project root: .
    Yaml found at: codecov.yml
==> Running gcov in . (disable via -X gcov)
==> Python coveragepy exists disable via -X coveragepy
    -> Running coverage xml
==> Searching for coverage reports in:
    + .
    -> Found 1 reports
==> Detecting git/mercurial file structure
==> Reading reports
    + ./coverage.xml bytes=484024
==> Appending adjustments
    http://docs.codecov.io/docs/fixing-reports
    -> No adjustments found
==> Gzipping contents
==> Uploading reports
    url: https://codecov.io
    query: branch=master&commit=31b9ae070e85c9b84c100e2861ec9eacf1e5190a&build=&build_url=&name=&tag=&slug=databricks%2Fkoalas&service=github-actions&flags=&pr=&job=
    -> Pinging Codecov
https://codecov.io/upload/v4?package=bash-20191211-b8db533&token=secret&branch=master&commit=31b9ae070e85c9b84c100e2861ec9eacf1e5190a&build=&build_url=&name=&tag=&slug=databricks%2Fkoalas&service=github-actions&flags=&pr=&job=
HTTP 400
Please provide the repository token to upload reports via `-t :repository-token`
```

### CODE REVIEW

File: .github/workflows/master.yml [hunk 1]
```diff
@@ -60,9 +60,9 @@ jobs:
         export SPARK_HOME="$SPARK_CACHE_DIR/spark-$SPARK_VERSION-bin-hadoop2.7"
         ./dev/lint-python
         ./dev/pytest
-    # Push the results back to codecov
-    - name: Update Codecov
-      run: bash <(curl -s https://codecov.io/bash)
+    - uses: codecov/codecov-action@v1
+      with:
+        token: ${{ secrets.CODECOV_TOKEN }}
```
Comment (harupy) [inline .github/workflows/master.yml]: Does this step work fine for a PR from a forked repo?

[The doc](https://help.github.com/en/actions/automating-your-workflow-with-github-actions/creating-and-using-encrypted-secrets#using-encrypted-secrets-in-a-workflow) says:

> With the exception of GITHUB_TOKEN, secrets are not passed to the runner when a workflow is triggered from a forked repository.

Comment (HyukjinKwon) [reply .github/workflows/master.yml]: Thanks @harupy. I think PR triggers codecov against the main repo. I'll investigate more tomorrow..

Comment (harupy) [reply .github/workflows/master.yml]: The secret doesn't seem to be passed to the runner.

Screen Shot 2020-02-10 at 12 01 00

Comment (harupy) [reply .github/workflows/master.yml]: or is it just hidden?

Comment (HyukjinKwon) [reply .github/workflows/master.yml]: I should add a Github Actions secret for `CODECOV_TOKEN`; however, I currently don't have the admin permission. Once that's set, everything should work fine.

**Comment by HyukjinKwon** [comment]
> Secrets are not passed to workflows that are triggered by a pull request from a fork.

Okay .. maybe I should merge and see the results ..

**Comment by HyukjinKwon** [comment]
Let me merge and see if it works.

**Comment by HyukjinKwon** [comment]
FYI @ueshin.

**Comment by HyukjinKwon** [comment]
okay .. seems working fine.

**Comment by harupy** [comment]
@HyukjinKwon @ueshin @itholic 

Tokenless Codecov action was released today.

https://github.com/codecov/codecov-action/issues/29#issuecomment-595062189To

**Comment by itholic** [comment]
Cool! Thanks for sharing @harupy :D

**Comment by HyukjinKwon** [comment]
Nice, thanks @harupy.

**Pull request merged**